### ISSUE TITLE
Date Picker's calendar is not launching on top of Dialog when Embedded in it.

### ISSUE THREAD

**Issue by problemsolverpk** [cerner/terra-framework#1282]
# Bug Report

## Description
Date Picker is opening the Calendar in background, when used in a Terra Notification Dialog.

## Steps to Reproduce
1. Create a Terra Notification Dialog
2. Embed DatePicker in the dialog using content prop
3. Open the calendar

See Code sandbox for issue - https://codesandbox.io/s/cocky-mclaren-wmqni?fontsize=14&hidenavigation=1&theme=dark
## Expected Behavior
The calendar should open on top of Dialog

## Environment

* Component Name and Version:
* Browser Name and Version:
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):

**Comment by pranav300** [comment]
Found the issue to be with the `Notification Dialog` z-index. We are providing the z-index to be 9000 for [abstract modal in Notification Dialog](https://github.com/cerner/terra-framework/blob/main/packages/terra-notification-dialog/src/NotificationDialog.jsx#L168). Whereas the default z-index for [Hookshot used by datepicker](https://github.com/cerner/terra-framework/blob/main/packages/terra-hookshot/src/HookshotContent.module.scss#L10) is  7001.

Planning to add a prop `zIndex` to notification dialog, so that the users can decide on which layer do they want the dialog to appear.

What do you guys think? @ryanthemanuel @mjhenkes  @ShettyAkarsh @dkasper-was-taken

**Comment by neilpfeiffer** [comment]
Thanks @problemsolverpk, the original design intent of the `terra-notification-dialog` was to primarily serve as a vehicle to deliver critical messages to users that require attention, and then direct them how to resolve within the normal base flow. 

Our z-index elevation strategy has been created for the current work like this:
6001: standard modals (`terra-modal-manager`)
7001: popups, form-select (the dropdown part), date-picker (popup calendar part), etc. 
9001: Notification Dialogs (e.g. errors, warnings, navigation checkpoints, time-out modals)


The most complex type of content we had originally planned for `terra-notification-dialog` was basic layout/markup (e.g. lists), text styling (bold/italics), and hyperlinks, instead of providing the ability to include additional form elements as the contents. Typically we'd recommend using a standard modal-manager if you need to have additional form details that need to be captured as part of validation or completing a flow, mainly because if you add form-validation and have an item incomplete, an error banner (terra-alert) should be injected as part of the content to accompany setting the field as `isInvalid`.

Depending on your exact flow you are trying to create or the design of the message content in your error dialog, using a replacement that keeps the form unit inline with the dialog could be an option. The [Date-Input](https://engineering.cerner.com/terra-ui/components/terra-date-input/date-input/date-input-field) uses a native select and won't have the same problem. Or the [Calendar Filter](https://engineering.cerner.com/terra-ui/components/terra-date-picker/date-picker/calendar-filter) could be an option, have the user pick a date from the calendar face-up and inline.

**Comment by neilpfeiffer** [comment]
Closing as not a bug, listed components are performing as expected. 

We will continue to keep these problems in mind as we continue work on [Terra Application ver. 2](https://github.com/cerner/terra-application/tree/terra-application-v2), which revisits modals the and z-index elevation strategy.
